### PR TITLE
Fix bug: can not calculate the size of pg_aoblkdir_xxxx before `create index on ao_table` is committed

### PR DESCRIPTION
**Pull request by zhrt123**

### CODE REVIEW

File: diskquota_utility.c [hunk 1]
```diff
@@ -1407,3 +1409,28 @@ diskquota_get_appendonly_aux_oid_list(Oid reloid, Oid *segrelid, Oid *blkdirreli
...
+				return atoi(&relname[13]);
+			else if (strncmp(relname, "pg_aocsseg", 10) == 0)
+				return atoi(&relname[11]);
+			else if (strncmp(relname, "pg_aoblkdir", 11) == 0)
+				return atoi(&relname[12]);
+		}
+		break;
+	}
+	return InvalidOid;
+}
```
Comment (higuoxing) [inline diskquota_utility.c:1436]: Nit: Append `\n` to the end of this file.

File: relation_cache.c [hunk 2]
```diff
@@ -424,19 +413,27 @@ add_auxrelation_to_relation_entry(Oid relid, DiskQuotaRelationCacheEntry *pentry
 	list_free(index_oids);
 }
 
-static void
+/*
+ * Returns true iff blkdirrelid missing.
+ * pg_aoblkdir_xxxx is created by `create index on ao_table`, which can not be
+ * searched by diskquota_get_appendonly_aux_oid_list() before index's creation 
+ * finish. By returning true to inform caller blkdirrelid missing, then caller
+ * will fetch blkdirrelid by traversing relation_cache.
```
Comment (higuoxing) [inline relation_cache.c:421]:
```suggestion
 * Returns true iff blkdirrelid is missing.
 * pg_aoblkdir_xxxx is created by `create index on ao_table`, which can not be
 * fetched by diskquota_get_appendonly_aux_oid_list() before index's creation 
 * finish. By returning true to inform the caller that blkdirrelid is missing,
 * then the caller will fetch blkdirrelid by traversing relation_cache.
```

**Review by higuoxing — APPROVED** [review]
:+1:  LGTM, but please wait until @xuebinsu is happy with this PR.

**Review by xuebinsu — DISMISSED** [review]
+1

File: diskquota_utility.c (hunk 1 above)
Comment (zhrt123) [reply diskquota_utility.c:1436]:
```suggestion
}

```

**Review by higuoxing — APPROVED** [review]

**Pull request merged**